---
title: 
output: 
  #rmarkdown::html_document:
   rmdformats::html_clean:
    toc: no
    highlight: tango
    #use_bookdown: false
    #css: styles.css

---
<div id="badge">
  <a href="https://docs.airr-community.org/en/stable/swtools/airr_swtools_standard.html">
    <img src="https://img.shields.io/static/v1?label=AIRR-C%20sw-tools%20v1&message=compliant&color=008AFF&labelColor=000000&style=plastic">
  </a>
</div>

<style type="text/css">
.main-container {
  max-width: 1800px;
  margin-left: auto;
  margin-right: auto;
}

h1 {
  color: #033c73;
   font-size: 24px;
}
h1.title {
  color: #033c73;
}
h2 {
  color: #033c73;
  font-size: 20px;
}
h3 {
   color: #033c73;
   font-size: 18px;
}
h4 {
   color: #022f5a;
}
h5 {
  color: #033c73;
}
h6 {
   color: #033c73;
}
body{ 
  font-size: 16px;
}
#badge {
  position: fixed;
  top: 10px;
  right: 10px;
}
</style>

<br><br>
**Shiny AnalyzAIRR** is the R shiny version of the AnalyzAIRR package developed for the analysis of bulk Ig/TCR repertoire datasets.

* It proposes an analytical routine starting with data exploration, leading to in-depth statistical comparisons with the aim of answering defined biological questions.

* It allows the calculation of a set of diversity measures and statistical metrics applicable at any level of granularity.

* It offers different types of data visualization and ready-to-publish graphics that can be easily personalized. 


# Data loading

## Upload a RepSeqExperiment object

A **RepSeqExperiment** object generated using the AnalyzAIRR R package and saved in an rds format can be used.

## Upload alignment files in AnalyzAIRR-compliant formats

**1. Specify an alignment file format**

  * **MiXCR**: https://github.com/milaboratory/mixcr

  * **immunoSEQ**: https://www.immunoseq.com/

  * **MiAIRR**: https://docs.airr-community.org/en/latest/datarep/rearrangements.html 
  
  * **Other**: Alignment files should contain the following required column names:

      - **sample_id**: Sample names
      
       - **V**: Variable gene name (IMGT nomenclature)
      
      - **J**: Joining gene name (IMGT nomenclature)
      
      - **ntCDR3**: Nucleotide CDR3 sequence
      
      - **aaCDR3**: Amino acid CDR3 sequence
      
      - **count**: the occurrence of each sequence within the sample
      
      **Note**: The order of the columns must be respected. Input files can contain           additional columns that will not be taken into account during the loading process.

**2. Specify the chain type to be analyzed (only one chain can be loaded at once)**

**3. Specify the paths to input files**

**4. Load the metadata**

  It is possible to provide a metadata if users wish to perform inter-group comparative analyses. 
  
  The metadata should be provided as a dataframe containing:

  * a column with the sample names that match the name of the alignment files and their order in the list.
  
  * any additional columns with relevant information for the analyses. Columns could encompass the experimental conditions, clinical variables, etc... 
  
  No specific column names nor order are required.

**5. Download the RepSeqExperiment object**

Once files are successfully downloaded, a RepSeqExperiment object is built which can downloaded in a rds format for subsequent usage.
<br><br>

