<div id="about" class="section level2">
<p><strong>Shiny AnalyzAIRR</strong> is a MiAIRR compliant web application for the analysis of bulk Ig/TCR repertoire datasets.</p>
<ul>
<li>It proposes an analytical routine starting with data exploration, leading to in-depth statistical comparisons with the aim of answering defined biological questions.</li>
<li>It allows the calculation of a set of diversity measures and statistical metrics applicable at any level of granularity.</li>
<li>It offers different types of data visualization and ready-to-publish graphics that can be easily personalized.</li>
</ul>
<p>Useful links:</p>
<ul>
<li><a href="https://github.com/vanessajmh/AnalyzAIRR" style="color: #3C8DBC">https://github.com/vanessajmh/AnalyzAIRR</a></li>
<li><a href="https://vanessajmh.github.io/AnalyzAIRR.github.io" style="color: #3C8DBC">https://vanessajmh.github.io/AnalyzAIRR.github.io</a></li>
<li><a href="https://github.com/vanessajmh/Shiny-AnalyzAIRR" style="color: #3C8DBC">https://github.com/vanessajmh/Shiny-AnalyzAIRR</a></li>
  
</ul>

 


<div id="how-to" class="section level2">
<h2>Data loading</h2>
<hr>
<h3> Upload a RepSeqExperiment object </h3>
<p>A <strong>RepSeqExperiment</strong> object generated using the AnalyzAIRR R package and saved in an rds format.</p>
<h3> Upload alignment files in AnalyzAIRR-compliant formats </h3>
<p><strong>1. Specify an alignment file format</strong></p>
<ul>
<li><strong>MiXCR</strong>: <a href="https://github.com/milaboratory/mixcr" style="color: #3C8DBC">https://github.com/milaboratory/mixcr</a></li>
<li><strong>immunoSEQ</strong>: <a href="https://www.immunoseq.com/" style="color: #3C8DBC">https://www.immunoseq.com/</a></li>
<li><strong>MiAIRR</strong>: <a href="https://docs.airr-community.org/en/latest/datarep/rearrangements.html" style="color: #3C8DBC">https://docs.airr-community.org/en/latest/datarep/rearrangements.html</a></li>
<li><strong>Other</strong>: Alignment files should contain the following required column names: </li>
<ul>
<li><strong>sample_id</strong>: Sample names</li>
<li><strong>CDR3nt</strong>: CDR3 nucleotide sequence</li>
<li><strong>CDR3aa</strong>: CDR3 amino acid sequence</li>
<li><strong>V</strong>: Variable gene name (IMGT nomenclature)</li>
<li><strong>J</strong>: Joining gene name (IMGT nomenclature)</li>
<li><strong>count</strong>: the occurrence of each sequence within the sample</li>
</ul>
<strong>Note</strong>: The order of the columns must be respected. Input files can contain additional columns that will not be taken into account during the loading process.
</ul>
<p><strong>2. Specify the chain type to be analyzed (only one chain can be loaded at once)</strong></p>
<p><strong>3. Specify the paths to input files</strong></p>
<p><strong>4. Load the metadata</strong></p>
It is possible to provide a metadata if users wish to perform inter-group comparative analyses.
<p>The metadata should be provided as a dataframe containing:</p>
<ul>
<li>a column with the sample names that match the name of the alignment files and their order in the list.</li>
<li>any additional columns with relevant information for the analyses. Columns could encompass the experimental conditions, clinical variables, etcâ€¦</li>
</ul>
<p>No specific column names nor order are required.</p>
<p><strong>5. Download the RepSeqExperiment object</strong></p>
Once files are successfully downloaded, a <strong>RepSeqExperiment</strong> object is built which can downloaded in a rds format for subsequent usage.

<h3> Start your analysis </h3>
<h4> Data manipulation </h4>
<p><strong>1. Visualize the data</strong></p>
<ul>
<li><strong>Assay</strong></li>
<li><strong>Metadata</strong></li>
<li><strong>Other data</strong></li>
<li><strong>History</strong></li>
</ul>
<p><strong>2. Filter the data</strong></p>
<ul>
<li><strong>Filter out a repertoire level count</strong></li>
<li><strong>Filter out unproductive sequences</strong></li>
<li><strong>Drop a sample</strong></li>
<li><strong>Extract shared sequences</strong></li>
<li><strong>Extract private sequences</strong></li>
<li><strong>Extract top sequences</strong></li>
</ul>
<p><strong>3. Normalize the data</strong></p>
<ul>
<li><strong>Down-sampling</strong></li>
<li><strong>Shannon-based normalization</strong></li>
</ul>
<p> <strong> Note </strong>: If multiple filters are applied on the <strong>RepSeqExperiment</strong> object, only the last applied filter will be taken into account. In order to perform multiple filtering, the object must be downloaded in a rds format and re-uploaded into the interface.</p>
<h4> Exploratory statistics </h4>
<p><strong>1. Basic statistics</strong></p>
<ul>
<li><strong>Metadata statistics</strong></li>
<li><strong>Detailed repertoire level statistics</strong></li>
</ul>
<p><strong>2. Diversity estimation</strong></p>
<ul>
<li><strong>Rarefaction analysis</strong></li>
<li><strong>Diversity indices</strong></li>
<li><strong>Renyi index</strong></li>
</ul>
<p><strong>3. Clonal distribution</strong></p>
<ul>
<li><strong>Per count interval</strong></li>
<li><strong>Per decreasing rank</strong></li>
</ul>
<h4> One-sample analysis </h4>
<ul>
<li><strong>Clonal distribution pr count intervals</strong></li>
<li><strong>V/J usage</strong></li>
<li><strong>V-J combination usage</strong></li>
<li><strong>CDR3 stacked spectractyping</strong></li>
<li><strong>CDR3 individual spectractyping</strong></li>
</ul>
<h4> Multi-sample analysis </h4>
<p><strong>1. Comparison of basic statistics</strong></p>
<ul>
<li><strong>Metadata statistics</strong></li>
<li><strong>Diversity indices</strong></li>
<li><strong>Renyi diversity</strong></li>
<li><strong>Count intervals</strong></li>
<li><strong>Rank distribution</strong></li>
<li><strong>V/J usage</strong></li>
</ul>
<p><strong>2. Similarity analysis</strong></p>
<ul>
<li><strong>Repertoire overlap</strong></li>
<li><strong>Repertoire correlation</strong></li>
<li><strong>Hierarchical clustering</strong></li>
<li><strong>Multidimensional scaling</strong></li>
</ul>
<p><strong>3. Differential analysis</strong></p>
<ul>
<li><strong>Volcano plot</strong></li>
<li><strong>Principal component analysis</strong></li>
</ul>
<p><strong>4. Spectratyping comparison</strong></p>

<p>Finally, you can visualize and download the final report of your analysis session. Do not hesitate to give it a title, specify your name(s) and to add text. Note that each table or plot can also be downloaded separately in pdf.</p>
<p>If you need help, do not hesitate to click on the question mark located at the top right of each panel to learn more about the function used.</p>

</div>


<style type="text/css">
.main-container {
  max-width: 1800px;
  margin-left: auto;
  margin-right: auto;
}

h1 {
  color: #033c73;
}
h1.title {
  color: #033c73;
}
h2 {
  color: #033c73;
}
h3 {
   color: #033c73;
}
h4 {
   color: #022f5a;
}
h5 {
  color: #033c73;
}
h6 {
   color: #033c73;
}
</style>